src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/SelectResultSet.java

```diff
@@ -226,23 +226,23 @@ private boolean containsWildcard(Query query) {
      */
     private List<Field> fetchFields(Query query) {
         Select select = (Select) query;
-        List<Field> fields;
+
         if (queryResult instanceof Aggregations) {
-            fields = select.getGroupBys().isEmpty() ? new ArrayList<>() : select.getGroupBys().get(0);
-            for (Field field : select.getFields()) {
-                if (field instanceof MethodField) {
-                    fields.add(field);
+            List<Field> groupByFields = select.getGroupBys().isEmpty() ? new ArrayList<>() : select.getGroupBys().get(0);
+
+            for (Field selectField : select.getFields()) {
+                if (selectField instanceof MethodField && !selectField.isScriptField()) {
```

Comment:
I assume what we want exactly here is de-duplicate. The `!selectField.isScriptField` check works on the assumption that there is no aggregation function implemented by painless script right? If so, is this true?

Reply:
It's even worse (because as per Elastic's DSL script aggregations can be for both Bucket and Metric aggregations), but it will require considerable redesign, because the process of transforming of parsed SQL into current domain model loses a lot of information. For example - for query like

select date_format(format), count(*)
from index
group by date_format(format)
order by date_format(format)

there's no easy way to understand that the date_format is actually the same, because at this stage we're operating only with transformed expressions.

Adding of SQLExpr into field information in this PR is related to future attempt to de-duplication that is necessary for order by.